src/map.cpp

```diff
@@ -2694,6 +2694,10 @@ std::pair<bool, bool> map::bash_ter_furn( const tripoint &p, const int str,
     bool smashed_something = false;
     bool success = false;
     int sound_volume = 0;
+    std::string soundfxid;
+    std::string soundfxvariant;
+    const auto &terid = ter_at( p );
```

Comment:
The names are a bit confusing, `ter_at` and `furn_at` return references to `ter_t` / `furn_t` objects, not ids of any kind.